source/lib/storage.js

```diff
@@ -74,3 +75,25 @@ export function setFavoriteBuildings(buildings: string[]) {
 export function getFavoriteBuildings(): Promise<Array<string>> {
 	return getItemAsArray(favoriteBuildingsKey)
 }
+
+/// MARK: Sis
+const courseDataKey = 'sis:course-data'
+export function setTermCourseData(term: number, courseData: Array<CourseType>) {
+	const key = courseDataKey + `:${term}:courses`
+	return setItem(key, courseData)
+}
+export function getTermCourseData(term: number): Promise<Array<CourseType>> {
+	const key = courseDataKey + `:${term}:courses`
+	// return getItemAsArray(key)
+	return AsyncStorage.getItem(`aao:${key}`).then(stored => JSON.parse(stored))
+}
+const termInfoKey = courseDataKey + ':term-info'
+export function setTermInfo(termData: Array<TermType>) {
+	return setItem(termInfoKey, termData)
+}
+export function getTermInfo(): Promise<Array<TermType>> {
+	// return getItemAsArray(termInfoKey)
+	return AsyncStorage.getItem(`aao:${termInfoKey}`).then(stored =>
```

Comment:
Please use `getItemAsArray` here, making sure to leave off the `aao:` prefix when you call it.

Reply:
Why didn't you want to use `getItemAsArray` here, again? Something about wanting it to return `null` if there are no terms known locally? I still think that returning an empty array is more appropriate there. (But if you disagree, the return type of this needs to become `Promise<?Array<TermType>>`, because it can currently return `null`.)

Reply:
I wasn't using `getItemAsArray` because it wasn't working after I cleared the storage and I couldn't figure out why..but I just figured it out and I'm a dummy so I'll change it to `getItemAsArray`...I'm still getting this flow error though when I do that:
```
Error: source/lib/storage.js:87
 87: 	return getItemAsArray(key)
     	       ^^^^^^^^^^^^^^^^^^^ Promise. This type is incompatible with the expected return type of
 85: export function getTermCourseData(term: number): Promise<Array<CourseType>> {
                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^ Promise
  Type argument `R` is incompatible:
     31: async function getItemAsArray(key: string): Promise<Array<*>> {
                                                             ^^^^^^^^ array type. Has some incompatible type argument with
     85: export function getTermCourseData(term: number): Promise<Array<CourseType>> {
                                                                  ^^^^^^^^^^^^^^^^^ array type
      Type argument `T` is incompatible:
         49: export function getHomescreenOrder(): Promise<Array<string>> {
                                                                 ^^^^^^ string. This type is incompatible with
         85: export function getTermCourseData(term: number): Promise<Array<CourseType>> {
                                                                            ^^^^^^^^^^ CourseType
```